x-pack/plugins/security_solution/public/detection_engine/rule_management_ui/components/maintenance_window_callout/maintenance_window_callout.test.tsx

```diff
@@ -0,0 +1,223 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { render, waitFor, cleanup } from '@testing-library/react';
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
+import { useAppToasts } from '../../../../common/hooks/use_app_toasts';
+import { useAppToastsMock } from '../../../../common/hooks/use_app_toasts.mock';
+import { MaintenanceWindowCallout } from './maintenance_window_callout';
+import { TestProviders } from '../../../../common/mock';
+import { fetchActiveMaintenanceWindows } from './api';
+
+jest.mock('../../../../common/hooks/use_app_toasts');
+
+jest.mock('./api', () => ({
+  fetchActiveMaintenanceWindows: jest.fn(() => Promise.resolve([])),
+}));
+
+const RUNNING_MAINTENANCE_WINDOW_1 = {
+  title: 'Maintenance window 1',
+  enabled: true,
+  duration: 1800000,
+  events: [{ gte: '2023-04-20T16:27:30.753Z', lte: '2023-04-20T16:57:30.753Z' }],
+  id: '63057284-ac31-42ba-fe22-adfe9732e5ae',
+  status: 'running',
+  expiration_date: '2024-04-20T16:27:41.301Z',
+  r_rule: {
+    dtstart: '2023-04-20T16:27:30.753Z',
+    tzid: 'Europe/Amsterdam',
+    freq: 0,
+    count: 1,
+  },
+  created_by: 'elastic',
+  updated_by: 'elastic',
+  created_at: '2023-04-20T16:27:41.301Z',
+  updated_at: '2023-04-20T16:27:41.301Z',
+  event_start_time: '2023-04-20T16:27:30.753Z',
+  event_end_time: '2023-04-20T16:57:30.753Z',
+};
+
+const RUNNING_MAINTENANCE_WINDOW_2 = {
+  title: 'Maintenance window 2',
+  enabled: true,
+  duration: 1800000,
+  events: [{ gte: '2023-04-20T16:47:42.871Z', lte: '2023-04-20T17:11:32.192Z' }],
+  id: '45894340-df98-11ed-ac81-bfcb4982b4fd',
+  status: 'running',
+  expiration_date: '2023-04-20T16:47:42.871Z',
+  r_rule: {
+    dtstart: '2023-04-20T16:47:42.871Z',
+    tzid: 'Europe/Amsterdam',
+    freq: 0,
+    count: 1,
+  },
+  created_by: 'elastic',
+  updated_by: 'elastic',
+  created_at: '2023-04-20T16:30:51.208Z',
+  updated_at: '2023-04-20T16:30:51.208Z',
+  event_start_time: '2023-04-20T16:47:42.871Z',
+  event_end_time: '2023-04-20T17:11:32.192Z',
+};
+
+const UPCOMING_MAINTENANCE_WINDOW = {
+  title: 'Upcoming maintenance window',
+  enabled: true,
+  duration: 45972,
+  events: [
+    { gte: '2023-04-21T10:36:14.028Z', lte: '2023-04-21T10:37:00.000Z' },
```

Comment:
Do we really need so many entries in the mock data? 😅

Reply:
This was my motivation: I mocked three MWs: two "running" and one "upcoming". I need at least two "running" to check the behaviour with multiple simultaneously running MWs. And I also need that "upcoming" one to make sure we don't show the callout for MWs that haven't started yet.

There's of course a question of should I even test these scenarios. Normally, we'd go through a test plan review first, then write test coverage. But given the short notice of this feature, I decided to cut a corner and do a team review of a test plan after FF. If we decide that it's not worth to test this scenario, I'll remove it and create a new PR.

Also each MW mock is a copy of an entry from an actual API response. I don't need most of these fields for my tests, but I decided to include the whole thing anyway for clarity.

What's your opinion on this?